### DISCUSSION TITLE
Can I change Free Air anomaly to gravity disturbance?

### DISCUSSION THREAD

**Discussion by prince-Ha** [fatiando/community#91] [Q&A] [answered]
Hi 
I'm learning a lot from fatiando. thank you

I became curious after reading the "Should geophysicists use the gravity disturbance or the anomaly?"  article.

The picture below is what I organized after reading the article. There was a difference in the gravitational effect between gravity disturbance and Free Air anomaly as much as the geoid height. 

![image](https://user-images.githubusercontent.com/69688086/214650243-5d10bda1-dead-463f-9df7-bfd588317e9f.png)

If I calibrate the gravitational effect corresponding to the height of the geoid to Free Air Anomaly, can I make a gravity disturbance and use it? The data I'm using is Free Air Anomaly  from Geoid, so I'm asking you to use gravity disturbance 

Thank you.

**Comment by santisoler** [comment]
Hi @prince-Ha. Thanks for opening this question.

Your derivations seems right, so using that last equation you could get from the free-air anomaly to an **approximation** of the gravity disturbance. I say **approximation** because the expression you have for the gravity disturbance in equation 1 is an approximation:

$$ \delta g_p = g_P - \gamma_p \simeq g_p - (\gamma_Q + \frac{\partial \gamma}{\partial h} h_p ) $$

The term between parenthesis is actually a first order Taylor approximation of the normal gravity on the point p. This approximation is actually not needed, [Li and Gotze (2001)](https://doi.org/10.1190/1.1487109) present a closed-form formula to compute the normal gravity on any external point. We have this formula [implemented in Boule](https://www.fatiando.org/boule/latest/user_guide/normal_gravity.html).

If you have details of how the free-air anomaly was applied to your data you can "undo" it and recover the observed gravity values $g_p$:

$$ g_p = \Delta g_F + (\gamma_Q + \frac{\partial \gamma}{\partial h} H_p) $$

And then you can compute the gravity disturbance using the analytical solution we have in Boule. Does it make sense?

**Reply by prince-Ha** [reply]
Thank you for your kind reply.
As you said, I will do the following to make the observed value.

1. make a observation height(H)
There is no height of observation point and only free air data exists. Therefore, using the topography data(at sea level), I will make the observation height to the topography height for the continent and zero for the sea.
2. Add observation height(H) and latitude to the bl.WGS84.normal_gravity()  to create the normal gravity(B).
3. observed gravity(A) = free air anomaly data + (B)
4.  Make a geometric height(h) = geoid height + (H)
5. Add observation geometric height(h) and latitude to the bl.WGS84.normal_gravity()  to create the normal gravity(C).
6. calculate the gravity disturbance = (A) - (C)

Isn't there a logical problem?
----------------------------------------------
7. in ocean is there no orthometic height?

Please check when you're not busy. Thank you.

**Reply by santisoler** [reply]
It sounds good, although I would compute the quantity you call (B) in a different way. If you use the Boule normal gravity function, you won't be removing the first order Taylor approximation that I assume they used to build the free-air anomaly: you are going to compute the analytical solution for the normal gravity at a given latitude and height. In addition, the `normal_gravity()` method takes **geometric heights**, and in your workflow you'll be passing a **orthometric height** to it.

I would instead compute B as follows:

$$ B = \gamma_Q + \frac{\partial \gamma}{\partial h} H_p $$

The first term can be computed with `bl.WGS84.normal_gravity(latitude=latitude, height=0)`: this way you get the normal gravity on a point located on the surface of the ellipsoid (zero geometric height).
The second term can easily be computed through the known free-air correction (Blakely, 1995, p.140):

$$ \frac{\partial \gamma}{\partial h} H_p  = -0.3086 \frac{\text{mGal}}{\text{m}} H_p $$

where $H_p$ is the orthometric height of the observation point (the one you define in your first point).

> in ocean is there no orthometic height?

An observation point located at the ocean surface has zero orthometric height. It's at sea level by definition and therefore located on the geoid, so zero orthometric height. But it's geometric height can be non-zero!

**Reply by prince-Ha** [reply]
thanks to you, I was able to convert free air anomaly to gravity disturbance

However another problem was occured at interpolation

I.
Artifacts occur while interpolating EIGEN6C4 geoid(ensaio, gridstep: 0.167 deg) data with gebco elevation grid(15").
I thought that interpolation could not be done in the latitude and longitude coordinate system, so I projected it with Mercator coordinates and interpolated it 

[code]
`gebco`
<img width="802" alt="image" src="https://user-images.githubusercontent.com/69688086/215274131-97f83cd5-ebab-4a92-b402-ef349cbcb25f.png">
`gebco_table = vd.grid_to_table(gebco)`
`geoid`
<img width="769" alt="image" src="https://user-images.githubusercontent.com/69688086/215274215-ce53ef1c-ccb3-4f50-9605-e43673396f21.png">

`projection = pyproj.Proj(proj = 'merc', lat_ts = gebco_table.lat.mean())`
`easting, northing = projection(gebco_table.lon.values, gebco_table.lat.values)`
<img width="500" alt="image" src="https://user-images.githubusercontent.com/69688086/215274359-7395cd69-12ab-4015-beb7-371d2dc6b8b1.png">
`gebco_proj = vd.project_grid(gebco.elevation, projection, method="nearest")`
`geoid_proj = vd.project_grid(geoid_eigen, projection, method="nearest")`
`geoid_proj_table = vd.grid_to_table(geoid_proj)`
`interpolator = vd.ScipyGridder(method = 'linear')`
`interpolator.fit((geoid_proj_table.easting, geoid_proj_table.northing), geoid_proj_table.geoid)`
`gebco_table = gebco_table.assign(geoid=interpolator.predict((gebco_table.easting_m, gebco_table.northing_m)))`
`gebco_table`


[at mercator interpolation]
![image](https://user-images.githubusercontent.com/69688086/215275436-d1e62343-f0f7-453e-abc4-eb19e60d513f.png)
cmap = jet

[geoid]

II.
Interpolation in latitude and longitude coordinates is artifact-free. Is it right to interpolate at longitude latitude coordinates?

[at longitude & latitude interpolation]
<img width="509" alt="image" src="https://user-images.githubusercontent.com/69688086/215275449-8b712a23-db40-4291-a28c-c51e58119f64.png">
cmap = turbo
[geoid]

It's cold. Be careful not to catch a cold.
thank you

**Reply by santisoler** [reply]
Hi @prince-Ha. You're right: interpolating on geographic coordinates using the current interpolators we have in Verde is not the best solution. But in this case where the region is quite long, projecting the grid also introduces high errors. Instead of having a projected and interpolated grid to then interpolate on the observation points, I would just run the interpolation on longitude latitude directly. I think this is the best compromise for now.

I hope at some point we can have nice interpolators for geographic coordinates that measure distance using Haversine distances rather than Euclidean distances.

**Reply by prince-Ha** [reply]
Thank you @santisoler 

Projections are bound to be accompanied by errors on a large scale....



I want to clarify the conventional terms. Help me.

In my country, gravity disturbance is still called free-air anomaly, so I'm really confused about the terminology.

So, the data provided by the site below is called Free Air Anomaly, but please ask if the actual concept is gravity disturbance.

https://www.generic-mapping-tools.org/remote-datasets/earth-faa.html

I think it's a disturbance because it's produced by satellite altimetry

**Reply by santisoler** [reply, marked as answer]
> In my country, gravity disturbance is still called free-air anomaly, so I'm really confused about the terminology.

They are actually two different quantities. I think you mean that people still use free-air anomaly as if they were gravity disturbances. If that so, chill out, it's not your country only, it's a worldwide issue 😅 

> So, the data provided by the site below is called Free Air Anomaly, but please ask if the actual concept is gravity disturbance.
> 
> https://www.generic-mapping-tools.org/remote-datasets/earth-faa.html
> 
> I think it's a disturbance because it's produced by satellite altimetry

I'm pretty certain that that is free-air anomaly and not gravity disturbance. But if you need to get satellite gravity disturbance, you can use [ICGEM Calculation Service](http://icgem.gfz-potsdam.de/calcgrid). In there you can choose which model you want to use (you can choose EGM2008, which is the one that GMT provides in the url you shared, but you can also choose a more update one like EIGEN-6C4), then choose which field you want to download (`gravity_disturbance_sa` for example) and give details of the grid where you want it and also the height above the ellipsoid.

You can use Harmonica's [load_icgem_gdf](https://www.fatiando.org/harmonica/latest/api/generated/harmonica.load_icgem_gdf.html#harmonica.load_icgem_gdf) function to load the file you'll download into an Xarray DataArray.

**Reply by prince-Ha** [reply]
I think you're good at teaching easily.

Thank you so much!!!
